src/org/ods/component/BuildOpenShiftImageStage.groovy

```diff
@@ -41,30 +41,32 @@ class BuildOpenShiftImageStage extends Stage {
     // Retrieve build status.
     def lastVersion = getLastVersion()
     if (!lastVersion) {
-      script.error "Could not get last version of BuildConfig '${context.componentId}'."
+      script.error "Could not get last version of BuildConfig '${componentId}'."
     }
-    def buildId = "${context.componentId}-${lastVersion}"
+    def buildId = "${componentId}-${lastVersion}"
     def buildStatus = getBuildStatus(buildId)
     if (buildStatus != 'complete') {
       script.error "OpenShift Build #${lastVersion} was not successful - status is '${buildStatus}'."
     }
-    context.addArtifactURI("OCP Build Id", buildId)
-
     def imageReference = getImageReference()
     script.echo "Build #${lastVersion} has produced image: ${imageReference}."
-    context.addArtifactURI("OCP Docker image", imageReference)
+    
+    context.addBuildToArtifactURIs (componentId, [ "buildId" : buildId, "image" : imageReference ])
+    
+    return ["buildId" : buildId, "image" : imageReference]
   }
 
     private String getImageReference() {
-      openShift.getImageReference(context.componentId, context.tagversion)
+      openShift.getImageReference(componentId, context.tagversion)
     }
 
     private String startAndFollowBuild() {
-      openShift.startAndFollowBuild(context.componentId, context.dockerDir)
+      def dockerContext = config.dockerfile ?: context.dockerDir
```

Comment:
I would move that to the `config` assembly above, and would name the option `dockerDir` instead of `dockerfile`: there are actually two things to customise for the image build: "build context" and "Dockerfile". Right now we only allow to customise the context - the file would need to be customised in the `BuildConfig` itself.